contrib/federations_tutorial/Federation.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+import os
+import random
+import sys
+import time
+import shutil
+import json
+import util
+from MultiSig import MultiSig
+from BlockSigning import BlockSigning
+from authproxy import AuthServiceProxy, JSONRPCException
+
+ELEMENTS_PATH = "../../src/elementsd"
+GENERATE_KEYS = True
+BLOCK_TIME = 60
+
+def main():
+    # GENERATE KEYS AND SINGBLOCK SCRIPT FOR FEDERATION BLOCK SIGNING
+    num_of_nodes = 3
+    num_of_sigs = 2
+    keys = []
+    signblockarg = ""
+
+    if GENERATE_KEYS:  # generate new signing keys and multisig
+        sig = MultiSig(num_of_nodes, num_of_sigs, ELEMENTS_PATH)
+        sig.generate()
+        keys = sig.privkeys
+        signblockarg = "-signblockscript={}".format(sig.multisig["redeemScript"])
+
+    else:   # use hardcoded keys and multisig
+        keys = ['Kz4aMbbciZrjrNBi4rwwrTemGKbj9qCdVVRKXAvyt1YK1mPFpRG8',
+                'Ky4A1C5xeTDSR4SWey6Lees9WHjGTfgVAK3vWngCjWQDFs5UqHx3',
+                'L5dPkjz7hzhGCfYAd19fDWsetdCCwUPRPtt3bLh7SSTrh3DFNY34']
+        signblockarg = "-signblockscript=522103d517f6e9affa60000a08d478970e6bbfa45d63b1967ed1e06" \
+                        "6dd46b802edb2a62102afc18e8a7ff988ca1ae7b659cb09a79852d301c2283e18cba1faf7a" \
+                        "0b020b1a22102edd8080e31f05c68cf68a97782ac97744e86ba19dfd3ba68e597f10868ee5bc453ae"
+
+    # INIT ELEMENTS FEDERATION NODES
+    elements_nodes = []
+    elements_datadirs = []
```

Comment:
Same here, there's no need to store the datadirs if we reuse the same random part of the path, then each node's path is calculated using "/tmp/RANDOM/node"+str(i).